tests/ttnn/unit_tests/gtests/ccl/kernels/fabric_worker_sender_multi_input.cpp

```diff
@@ -0,0 +1,222 @@
+    const size_t dest_noc_y = (dest_noc_address >> (NOC_ADDR_LOCAL_BITS + NOC_ADDR_NODE_ID_BITS)) & ((1 << NOC_ADDR_NODE_ID_BITS) - 1);
+    const size_t packet_size = page_size + sizeof(tt::fabric::PacketHeader);
+
+    auto packet_addr = get_read_ptr(cb_id);
+    auto &packet_header = *reinterpret_cast<tt::fabric::PacketHeader*>(packet_addr);
+    if constexpr (mcast_mode) {
+        packet_header.to_write()
+            .to_chip_multicast(tt::fabric::MulticastRoutingCommandHeader{config.mcast.distance, config.mcast.range})
+            .to_noc_unicast(tt::fabric::NocUnicastCommandHeader{
+                dest_addr,
+                (pages_to_send * page_size) + sizeof(tt::fabric::PacketHeader),
+                static_cast<uint8_t>(dest_noc_x),
+                static_cast<uint8_t>(dest_noc_y)
+            });
+        packet_header.reserved2 = 0x1111; // debug only
+    } else {
+        packet_header.to_write()
+            .to_chip_unicast(tt::fabric::UnicastRoutingCommandHeader{config.unicast.distance})
+            .to_noc_unicast(tt::fabric::NocUnicastCommandHeader{
+                dest_addr,
+                (pages_to_send * page_size) + sizeof(tt::fabric::PacketHeader),
+                static_cast<uint8_t>(dest_noc_x),
+                static_cast<uint8_t>(dest_noc_y)
+            });
+        packet_header.reserved2 = 0x1111; // debug only
+    }
+
+    uint64_t buffer_address = sender.edm_buffer_addr + (*sender.buffer_index_ptr * (sender.buffer_size_bytes + sizeof(eth_channel_sync_t)));
+    sender.send_payload_blocking_from_address(packet_addr, packet_size);
+    noc_async_writes_flushed();
+    // }
+}
+
+template <typename AddrGen>
+void non_blocking_read_and_forward(size_t &current_page_in, uint32_t cb_id, const AddrGen &dest_addr_gen, tt::fabric::WorkerToFabricEdmSender &sender, transmit_config const& config, uint32_t page_size, uint32_t total_pages_to_send, uint32_t num_pages_per_send) {
+    uint32_t pages_to_send = std::min<uint32_t>(num_pages_per_send, total_pages_to_send - current_page_in);
+    if (!cb_pages_available_at_front(cb_id, pages_to_send)) {
+        return;
+    }
+
+    current_page_in += num_pages_per_send;
+    cb_wait_front(cb_id, pages_to_send);
+    forward_to_fabric_from_cb(
+        total_pages_to_send,
+        sender,
+        cb_id,
+        config,
+        page_size,
+        dest_addr_gen,
+        num_pages_per_send,
+        current_page_in
+        );
+    cb_pop_front(cb_id, pages_to_send);
+}
+
+// Worker core - Data Movement Writer -> Sends to Erisc Data Mover (sender side).
+// -> takes input from local cb and pushes to erisc L1
+void kernel_main() {
+
+    // Test doesn't support multiple pages per send yet since we are writing
+    // to interleaved which will never have subsequent pages on the same core
+    // (and hence, able to share a packet header)
+    constexpr uint32_t num_pages_per_send = 1;//get_compile_time_arg_val(0);
```

Comment:
Why did the compile time var get replaced with a constant?

Reply:
please see the comment :)

Reply:
Ah, never mind then. I guess in that case its good to "save" compile val 0 as you did for when the support comes in